### DISCUSSION TITLE
How to send server side message from python background task?

### DISCUSSION THREAD

**Discussion by pkumar2philips** [miguelgrinberg/Flask-SocketIO#1810] [Q&A]
It seems to me that server can only send message if its mentioned with @socketio.event/@socketio.on over a function. if we try to send/emit message outside this scope then its not working. 

Do we have any solution for this?

**Comment by miguelgrinberg** [comment]
Don't know why you reached that conclusion. Emitting from the server without a previous client event is fully supported. It is even in the example application: https://github.com/miguelgrinberg/Flask-SocketIO/blob/main/example/app.py#L25

**Reply by pkumar2philips** [reply]
Thanks for the reply. I did see this and it seems to me that we can only use the background_task for achieving this.  we need to use some kind of queue to add all message from different part of application and then use this background task for sending to client. 

We are also experiencing some application lag after using the background task. Does background_task use the more resources since we are running this code in while loop.

**Reply by miguelgrinberg** [reply]
I think your confusion with this might be with regards to the "different part of the application". What do you mean by that exactly? If you are using an async framework, then all you have is tasks, so a background task that you start, versus one on which another part of your application runs should be exactly the same in terms of being able to emit. The problem that you have could be that you are blocking the async application with some things you are doing. That would explain emits not working or getting delayed.
